parlai/tasks/wrapper/agents.py

```diff
@@ -18,43 +18,51 @@
 
 import copy
 
+from abc import ABC, abstractmethod
 from parlai.core.agents import create_agent_from_shared
 from parlai.core.opt import Opt
 from parlai.core.teachers import create_task_agent_from_taskname, Teacher
 
 
-class LabelToTextTeacher(Teacher):
+class AbstractWrapperTeacher(Teacher, ABC):
     """
-    Teacher that will shift message['labels'][0] into message['text'] for whatever task
-    is specified with --label-to-text-task.
-
-    Because the dialogue history is effectively overwritten by this action, all episodes
-    will be flattened into one example each.
+    Abstract teacher that will wrap around another teacher and allow for manipulating
+    the fields returned by the inner teacher.
     """
 
     @classmethod
-    def add_cmdline_args(cls, argparser):
-        # Teacher has no args
-        parser = argparser.add_argument_group('LabelToText args')
-        parser.add_argument(
-            '-l2tt',
-            '--label-to-text-task',
+    def add_cmdline_args(cls, parser):
+        agent = parser.add_argument_group('AbstractWrapper args')
+        agent.add_argument(
+            '-wt',
+            '--wrapper-task',
```

Comment:
Ah, so there's one last annoying step needed here - since you're changing the name of an existing flag, you'll need to change this name in all of the places in the ParlAI code where this flag already exists, or otherwise that code will break once this PR is merged in. Fortunately this should be pretty simple - you can search for all places in ParlAI base and ParlAI-Internal where the strings `--label-to-text-task` and `label_to_text_task` appear, and change them to `--wrapper-task` and `wrapper_task`, respectively. (Your IDE should have an easy way to search over the entire repos - lemme know if it's not working for you :) )